### PR TITLE
Compile bzlib as shared on Linux but static on OS X and other platforms.  Thanks @bgruening and @peterjc!

### PR DESCRIPTION
**Pull request by gregvonkuster**

### CODE REVIEW

**Comment by peterjc** [comment]
Why not go further and use the XML architecture specific actions, rather than a bash if?

I think you should add an explicit fail if the OS is neither Linux or Mac OS X.

**Comment by gregvonkuster** [comment]
@peterjc Yes, you're right.  A better fix coming soon!

**Comment by bgruening** [comment]
@gregvonkuster this is what I meant with 'Galaxy-specific-OS-validation' :)

**Comment by gregvonkuster** [comment]
@bgruening Hopefully my latest fix should be better.  I've tested on both Linux and OS X.

File: packages/package_bzlib_1_0/tool_dependencies.xml [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
+            <actions_group>
+                <actions os="linux" architecture="x86_64">
+                    <!-- first action is always downloading -->
+                    <action type="download_by_url">http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz</action>
+                    <action type="shell_command">
+                        <![CDATA[
+                            make -f Makefile-libbz2_so &&
+                            make &&
+                            make install PREFIX=$INSTALL_DIR/bzlib &&
+                            cp bzip2-shared $INSTALL_DIR/bzlib/bin/bzip2 &&
```
Comment (yhoogstrate) [inline packages/package_bzlib_1_0/tool_dependencies.xml]: You could add ```make test &&```. This runs 6 tests cases to confirm the tools works.

Comment (nsoranzo) [inline packages/package_bzlib_1_0/tool_dependencies.xml]: There is no policy here, but I don't usually include `make test` in installation recipes.

**Comment by bgruening** [comment]
LGTM! Thanks @gregvonkuster to fix this and endure my nitpicking :)
@yhoogstrate we had plans to add a proper test framework to version 2 of the `tool_depenencies` but I don't think this will happen ...
That said, feel free to add tests into your dependencies I would not force it for contributions as we have no solid infrastructure for it now.

**Pull request merged**